### PR TITLE
search api completed

### PR DESCRIPTION
**Pull request by kenny0622**
I've completed search api. Please test and feel free to let me know if there is any problem.

### CODE REVIEW

**Comment by kenny0622** [comment]
bug has been fixed

File: app/models/doctor.rb [hunk 1]
```diff
@@ -9,6 +9,9 @@ class Doctor < ActiveRecord::Base
   has_many :divisions, :through => :div_hosp_doc_ships
   has_many :comments
   belongs_to :area
+  attr_accessor :hospital_id
+  attr_accessor :hospital_name
+  attr_accessor :doctor_id
```
Comment (StevenKo) [inline app/models/doctor.rb]: 這裡 doctor_id 是一定要的嗎？我測是可以不用？！

File: app/controllers/import_data_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    repository.type = 'hospital'
+    repository.create_index! force: true
+    #repository.delete_index?
+    hospitals = Hospital.all
+    hospitals.each do |v|
+      repository.save(v.as_json)
+    end
+    render :json => "success"
+  end  
+end
```
Comment (StevenKo) [inline app/controllers/import_data_controller.rb]: 這個檔案用不到？可以刪了

File: app/controllers/api/v1/search_controller.rb [hunk 3]
```diff
@@ -3,15 +3,21 @@ def search_doctors
...
       doctors = []
     else
-      doctors = Doctor.search params[:q]
+      repository = Elasticsearch::Persistence::Repository.new
+      repository.index = "doctors_index"
+      repository.type = 'doctor'
+      doctors = repository.search({query: { match: { name: params[:q] } }, size: 30})   
     end
-    render :json => doctors
+    render :json => doctors.to_json(:methods => [:hospital_name, :hospital_id])
```
Comment (StevenKo) [inline app/controllers/api/v1/search_controller.rb]: 這個只 select 這些欄位嗎？
doctors.id,doctors.name,hospitals.name as hospital_name,hospitals.id as hospital_id,doctors.latitude,doctors.longitude,doctors.comment_num,doctors.recommend_num,doctors.avg

Comment (StevenKo) [inline app/controllers/api/v1/search_controller.rb]: 因為如果全部的欄位都select， android parse 要再改寫

File: app/controllers/api/v1/search_controller.rb [hunk 4]
```diff
@@ -3,15 +3,21 @@ def search_doctors
...
     if params[:q].nil?
       hospitals = []
     else
-      hospitals = Hospital.search params[:q]
+      repository = Elasticsearch::Persistence::Repository.new
+      repository.index = "hospitals_index"
+      repository.type = 'hospital'
+      hospitals = repository.search({query: { match: { name: params[:q] } }, size: 30})
     end
     render :json => hospitals
```
Comment (StevenKo) [inline app/controllers/api/v1/search_controller.rb]: select('id,name,address,grade,latitude,longitude,comment_num,recommend_num,avg') 
這些欄位就好

File: app/models/doctor.rb [hunk 5]
```diff
@@ -9,6 +9,8 @@ class Doctor < ActiveRecord::Base
   has_many :divisions, :through => :div_hosp_doc_ships
   has_many :comments
   belongs_to :area
+  attr_accessor :hospital_id
+  attr_accessor :hospital_name
```
Comment (StevenKo) [inline app/models/doctor.rb]: 發現這樣寫之後，之前的api就壞了, @@

File: app/controllers/api/v1/search_controller.rb [hunk 6]
```diff
@@ -3,16 +3,22 @@ def search_doctors
...
       doctors = []
     else
-      doctors = Doctor.search params[:q]
+      repository = Elasticsearch::Persistence::Repository.new
+      repository.index = "doctors_index"
+      repository.type = 'doctor'
+      doctors = repository.search({query: { match: { name: params[:q] } }, size: 30})   
     end
-    render :json => doctors
+    render :json => doctors.to_json({:method => [:hospital_name, :hospital_id], :only => [:id, :name, :hospital_name, :hospital_id, :latitude, :longitude, :comment_num, :recommend_num, :avg]})
```
Comment (StevenKo) [inline app/controllers/api/v1/search_controller.rb]: 我在 local 測，這樣會少:hospital_name, :hospital_id ?? 怪！

**Pull request merged**